ext/simplexml/simplexml.c

```diff
@@ -2567,7 +2567,7 @@ PHP_FUNCTION(simplexml_import_dom)
 	nodep = php_libxml_import_node(node);
 
 	if (!nodep) {
-		zend_argument_type_error(1, "must be of type SimpleXMLElement|DOMNode, %s given", zend_zval_value_name(node));
+		zend_argument_value_error(1, "must be a valid XML node");
```

Comment:
Not sure if this needs to be a value error, it is consistent with ext/xsl; but perhaps those should be type errors too instead?

Reply:
Having it be a type error probably makes more sense? I don't imagine it is possible to introduce an interface in ext/libxml that extensions that hook into need to implement?

Reply:
Yeah, I'd also vote for a `TypeError`

Reply:
Alright, I pushed that as an extra commit to this PR.